### PR TITLE
Lra 373 broadcast attention message to header

### PR DESCRIPTION
**Pull request by britaumich**
Broadcast "Attention Needed" messages

- attention model created to create "Attention needed" or "" message to broadcast it from the model
- attention_channel.js updates a Dashboard
- lib/tasks/check_rooms_status.rake - a task that checks states and broadcasts "attention needed" or clear message

Lamp Hours filter

- index.html.erb displays a filter
- rooms_comtroler - creates @rooms
- autosubmit_controller.js - validated input values
- config/initializers/constants.rb - min and max numbers for lamp hours 

_room_card has updated information about rooms

Please let me know if you have questions

### CODE REVIEW

File: lib/tasks/check_rooms_status.rake [hunk 1]
```diff
@@ -0,0 +1,37 @@
+desc "This will find rooms that need attention"
+task check_rooms_status: :environment do
```
Comment (rsmoke) [inline lib/tasks/check_rooms_status.rake:2]: When does this rake task get called?

File: lib/tasks/check_rooms_status.rake [hunk 2]
```diff
@@ -0,0 +1,37 @@
+desc "This will find rooms that need attention"
```
Comment (rsmoke) [inline lib/tasks/check_rooms_status.rake:1]: when or what triggers this task

**Review by rsmoke — CHANGES_REQUESTED** [review]
delete commented or unused code. 
This is certainly a big step and a major improvement. I would like to walk through the PR with  you.

File: lib/tasks/check_rooms_status.rake (hunk 1 above)
Comment (britaumich) [reply lib/tasks/check_rooms_status.rake:2]: It's called from the cron job:
```
litx-brita:avmonitor brita$ crontab -l
*/1 * * * * /bin/bash -l -c 'export PATH="/Users/brita/.rbenv/bin:$PATH"; eval "$(rbenv init -)"; cd /Users/brita/avmonitor && RAILS_ENV=development bundle exec ./bin/rake check_rooms_status' >> /Users/brita/cron_logs/cron.txt 2>&1
```

File: app/views/attentions/_attention.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<span>
```
[the remark this answers is not in the document]
Comment (britaumich) [reply app/views/attentions/_attention.html.erb:1]: oh no, my usual error again - fixed

File: app/javascript/controllers/autosubmit_controller.js [hunk 4]
```diff
@@ -1,18 +1,57 @@
...
-      this.statusTarget.textContent = 'Updating...'
-      Turbo.navigator.submitForm(this.formTarget)
-      this.sidebarTarget.classList.toggle('-translate-x-full')
+      var min_lamp_hour = parseInt(this.min_lamp_hourTarget.value)
+      var max_lamp_hour = parseInt(this.max_lamp_hourTarget.value)
+
+      if (Number.isNaN(min_lamp_hour)) { min_lamp_hour = 0 }
+      if (Number.isNaN(max_lamp_hour)) { max_lamp_hour = 0 }
+
+      // if (min_lamp_hour != 0 && min_lamp_hour > MIN || max_lamp_hour != 0 && max_lamp_hour < MAX) {
```
[the remark this answers is not in the document]
Comment (britaumich) [reply app/javascript/controllers/autosubmit_controller.js]: done

File: lib/tasks/check_rooms_status.rake (hunk 2 above)
Comment (britaumich) [reply lib/tasks/check_rooms_status.rake:1]: t's called from the cron job:
```
litx-brita:avmonitor brita$ crontab -l
*/1 * * * * /bin/bash -l -c 'export PATH="/Users/brita/.rbenv/bin:$PATH"; eval "$(rbenv init -)"; cd /Users/brita/avmonitor && RAILS_ENV=development bundle exec ./bin/rake check_rooms_status' >> /Users/brita/cron_logs/cron.txt 2>&1
```

File: app/views/attentions/_attention.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (rsmoke) [reply app/views/attentions/_attention.html.erb:1]: I am not seeing the fix here

**Comment by britaumich** [comment]
> delete commented or unused code. This is certainly a big step and a major improvement. I would like to walk through the PR with you.

With pleasure - I'll review the code with you as soon as you have time for that.

File: lib/tasks/check_rooms_status.rake (hunk 2 above)
Comment (rsmoke) [reply lib/tasks/check_rooms_status.rake:1]: can you add a comment in the code that describes the cron job. you set on your machine. IE. the output of the contab -l

**Review by rsmoke — CHANGES_REQUESTED** [review]

**Comment by britaumich** [comment]
> can you add a comment in the code that describes the cron job. you set on your machine. IE. the output of the contab -l

Too many comments. Here it is again:
It's called from the cron job:
```
litx-brita:avmonitor brita$ crontab -l
*/1 * * * * /bin/bash -l -c 'export PATH="/Users/brita/.rbenv/bin:$PATH"; eval "$(rbenv init -)"; cd /Users/brita/avmonitor && RAILS_ENV=development bundle exec ./bin/rake check_rooms_status' >> /Users/brita/cron_logs/cron.txt 2>&1
```

File: lib/tasks/check_rooms_status.rake (hunk 2 above)
Comment (britaumich) [reply lib/tasks/check_rooms_status.rake:1]:
```
litx-brita:avmonitor brita$ crontab -l
*/1 * * * * /bin/bash -l -c 'export PATH="/Users/brita/.rbenv/bin:$PATH"; eval "$(rbenv init -)"; cd /Users/brita/avmonitor && RAILS_ENV=development bundle exec ./bin/rake check_rooms_status' >> /Users/brita/cron_logs/cron.txt 2>&1
```

**Review by rsmoke — COMMENTED** [review]
Monday we will walk through this PR. Using ActionCable is cool stuff

**Review by rsmoke — COMMENTED** [review]
Need to √ for If not Attention message in system in RAKE task

**Review by rsmoke — APPROVED** [review]

**Pull request merged**